tests/Functional/ControllerTest.php

```diff
@@ -0,0 +1,211 @@
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Kernel;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollectionBuilder;
+use Violines\RestBundle\Tests\Stubs\MimeTypes;
+use Violines\RestBundle\ViolinesRestBundle;
+
+/**
+ * @coversNothing
+ */
+final class ControllerTest extends TestCase
+{
+    private TestKernel $app;
+
+    protected function setUp(): void
+    {
+        $this->deleteTempDir();
+
+        $this->app = new TestKernel();
+        $this->app->boot();
+    }
+
+    protected function tearDown(): void
+    {
+        $this->deleteTempDir();
+    }
+
+    public function testReturnsOne(): void
+    {
+        $request = Request::create('/returnsOne');
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertSame(MimeTypes::APPLICATION_JSON, $response->headers->get('Content-Type'));
+        self::assertJsonStringEqualsJsonString(<<<JSON
+            {"from":"Forest"}
+       JSON, $response->getContent());
+    }
+
+    public function testReturnsMany(): void
+    {
+        $request = Request::create('/returnsMany');
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertSame(MimeTypes::APPLICATION_JSON, $response->headers->get('Content-Type'));
+        self::assertJsonStringEqualsJsonString(<<<JSON
+            [
+                {"from":"Jenny"},
+                {"from":"Forest"}
+            ]
+       JSON, $response->getContent());
+    }
+
+    public function testReconstitutesOneFromRequest(): void
+    {
+        $submitted = <<<JSON
+            {"from":"Jenny"}
+        JSON;
+
+        $request = Request::create('/reconstitutesOne', Request::METHOD_POST, [], [], [], [], $submitted);
+        $request->headers->set('Content-Type', MimeTypes::APPLICATION_JSON);
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertJsonStringEqualsJsonString($submitted, $response->getContent());
+    }
+
+    public function testReconstitutesMultipleFromRequest(): void
+    {
+        $submitted = <<<JSON
+            [
+                {"from":"Jenny"},
+                {"from":"Forest"}
+            ]
+        JSON;
+
+        $request = Request::create('reconstitutesMany', Request::METHOD_POST, [], [], [], [], $submitted);
+        $request->headers->set('Content-Type', MimeTypes::APPLICATION_JSON);
+        $request->headers->set('Accept', MimeTypes::APPLICATION_JSON);
+
+        $response = $this->app->handle($request);
+
+        self::assertJsonStringEqualsJsonString($submitted, $response->getContent());
+    }
+
+    private function deleteTempDir()
```

Comment:
you could check, if we could have the cache just in memory, e.g. with vfsStream: https://github.com/bovigo/vfsStream

then we wouldn't need to clean up, the gc would take care of it.

Reply:
I don't think this works with symfony. Do you have any example out there?

As far as I understand https://symfony.com/doc/current/reference/configuration/kernel.html#cache-directory only the cache directory is configurable, but not the Cache itself.

See [here](https://github.com/symfony/symfony/blob/10d869d8357de8005a2a0feacb36020db1a63b7f/src/Symfony/Component/HttpKernel/Kernel.php#L429-L586)

Reply:
yea, maybe. I think I will try it out after you PR is merged. Sounds like a cool challenge. :P Will let you know, if I succeed!